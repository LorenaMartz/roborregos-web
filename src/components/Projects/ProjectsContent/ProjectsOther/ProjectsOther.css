.projects-other {
    height: 100%;
    position: relative;
    background: white;
  }

  .other-info {
      padding: 10vw;
      padding-top: 10vh;
      padding-bottom: 5vh;
  }
  
  .description-other {
    font-size: 3vh;
    text-transform: none;
    padding-top: 2%;
    padding-bottom: 0%;
    color: black;
  }
  
  .title-text-other {
    font-size: 7vh;
    text-transform: none;
    color: black;
  }

  .project-title {
    font-size: 3vh;
    text-align: center;
    color: black;
    height: 5vh;
  }

  .project-grid-tile {
    background-color: white;
    border-width: 1vw;
    border-style: solid;
    border-color: black;
  }

  .projects-other-grid {
    /* background-color: chartreuse; */
    background: linear-gradient(0deg, black 50%, white 50%);
    padding: 0;
    width: 100%;
    display: block;
    cursor: pointer;
    overflow: hidden;
    padding-left: 8vw;
    padding-right: 8vw;
    padding-bottom: 5vh;
  }

  .other-project-image {
    background-color: rgba(0, 0, 0, 0.5);
    display: block;
    margin-left: auto;
    margin-right: auto;
    height: 40vh;
  }

  
  