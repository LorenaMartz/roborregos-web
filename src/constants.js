export const LARGE_WIDTH = 1200
export const MEDIUM_WIDTH = 992
export const SMALL_WIDTH = 768
export const XSMALL_WIDTH = 592
export const MOBILE_WIDTH = 414
