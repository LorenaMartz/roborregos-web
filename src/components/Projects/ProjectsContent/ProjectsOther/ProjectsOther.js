// @flow
import React, { Component } from 'react'
import './ProjectsOther.css'
import GridList from '@material-ui/core/GridList'
import GridListTile from '@material-ui/core/GridListTile'
import { Row, Col, Container } from 'react-bootstrap'
import { SMALL_WIDTH, LARGE_WIDTH } from '../../../../constants'

type Project = {
  title: string,
  image: string,
  wiki: string,
  background: string,
  color: string
};

type Props = {
  projects: Array<Project>
};

type State = {
  large_view: boolean
};

/** Component class of Projects' header. */
class ProjectsOther extends Component<Props, State> {
  projects: Array<Project>;

  /**
   * Class constructor
   * @param {list} props: List of projects data.
   */
  constructor(props: Props) {
    super(props)
    this.largeView = this.largeView.bind(this)
    this.viewSizeLarge = this.viewSizeLarge.bind(this)
    this.updateView = this.updateView.bind(this)
    this.smallView = this.smallView.bind(this)

    this.projects = props.projects
    this.state = {
      large_view: this.viewSizeLarge(),
    }
  }

  componentDidMount() {
    window.addEventListener('resize', this.updateView)
  }

  learnMoreCallback = (link: string) => {
    window.open(link)
  }

  updateView = () => {
    this.setState({
      large_view: this.viewSizeLarge(),
    })
  }

  viewSizeLarge = () => {
    if (window.innerWidth > SMALL_WIDTH) {
      return true
    }
    return false
  }

  /**
  * Parses path to projects image
  * @param {string} imgPath: Path to project's image/photo.
  * @return {path}
  */
  tryRequire = (imgPath: string) => {
    try {
      /* eslint-disable import/no-dynamic-require */
      /* eslint-disable global-require */
      return require(`../../../../images/projects/${imgPath}`)
    } catch (err) {
      return imgPath
    }
  }

  generateSmallCard = (project: Project, index: number) => {
    if (index % 2) {
      return (
        <Container
          id={`other-project-small-${project.image}`}
          className="other-projects-small"
          style={{ backgroundColor: project.background }}
          onClick={() => this.learnMoreCallback(project.wiki)}
        >
          <Row>
            <Col xs={5}>
              <div className="project-title" style={{ color: project.color }}>
                <p>
                  {project.title}
                </p>
              </div>
            </Col>
            <Col xs={7}>
              <img
                className="other-project-image"
                src={this.tryRequire(`${project.image}.jpg`)}
                alt={project.title}
              />
            </Col>
          </Row>
        </Container>
      )
    }
    return (
      <Container
        id={`other-project-small-${project.image}`}
        className="other-projects-small"
        style={{ backgroundColor: project.background }}
        onClick={() => this.learnMoreCallback(project.wiki)}
      >
        <Row>
          <Col xs={7}>
            <img
              className="other-project-image"
              src={this.tryRequire(`${project.image}.jpg`)}
              alt={project.title}
            />
          </Col>
          <Col xs={5}>
            <div className="project-title" style={{ color: project.color }}>
              <p>
                {project.title}
              </p>
            </div>
          </Col>
        </Row>
      </Container>
    )
  }

  largeView = () => (
    <div className="projects-other">
      <Row>
        <div className="other-info">
          <h2 className="title-text-other">
            Other projects
          </h2>
          <div className="description-other">
            <p>
              Check out other interesting tech projects developed by the community!
            </p>
          </div>
        </div>
      </Row>
      <Row>
        <div className="projects-other-grid">
          <GridList
            cellHeight="auto"
            cols={(window.innerWidth > LARGE_WIDTH) ? 4 : 2}
            spacing={20}
          >
            {this.projects.map((project, index) => (
              <GridListTile
                id={`other-project-${project.image}`}
                key={index}
                onClick={() => this.learnMoreCallback(project.wiki)}
              >
                <div
                  className="other-project-tile-content"
                  style={{ backgroundColor: project.background }}
                >
                  <img
                    className="other-project-image"
                    src={this.tryRequire(`${project.image}.jpg`)}
                    alt={project.title}
                    // style={{ backgroundColor: project.background }}
                  />
                  <div className="project-title">
                    <p>
                      {project.title}
                    </p>
                  </div>
                </div>
              </GridListTile>
            ))}
          </GridList>
        </div>
      </Row>
    </div>
  )

  smallView = () => (
    <div className="projects-other">
      <div className="other-info">
        <h2 className="title-text-other">
          Other projects
        </h2>
        <div className="description-other">
          <p>
            Check out other interesting tech projects developed by the community!
          </p>
        </div>
      </div>
      <>
        {this.projects.map((project, index) => this.generateSmallCard(project, index))}
      </>
    </div>
  )

  /**
 * Renders Responsive view of Projects's Other section.
 * @return {renderized_component} Other section with proyect grid or rows.
 */
  render() {
    const { large_view } = this.state
    if (large_view) return this.largeView()
    return this.smallView()
  }
}

export default ProjectsOther
