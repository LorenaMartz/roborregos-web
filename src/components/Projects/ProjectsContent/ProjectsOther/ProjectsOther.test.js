// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ProjectsOther from './ProjectsOther'
import { LARGE_WIDTH, SMALL_WIDTH } from '../../../../constants'

let wrapper
configure({ adapter: new Adapter() })

const other = [
  {
    title: 'other_project1',
    wiki: 'https://github.com/1',
    image: 'other_project1',
    background: '#FFFFFF',
    color: 'black',
  },
  {
    title: 'other_project2',
    wiki: 'https://github.com/2',
    image: 'other_project2',
    background: '#FFFFFF',
    color: 'black',
  },
]

beforeEach(() => {
  wrapper = shallow(
    <ProjectsOther projects={other} />,
  )
})

it('<ProjectsOther> renders large view components correctly', () => {
  expect(wrapper.find('.projects-other').length).toBe(1)
  expect(wrapper.find('.title-text-other').length).toBe(1)
  expect(wrapper.find('.other-info').length).toBe(1)
  expect(wrapper.find('.description-other').length).toBe(1)
  expect(wrapper.find('.projects-other-grid').length).toBe(1)
  expect(wrapper.find('.other-project-image').length).toBe(2)
  expect(wrapper.find('.project-title').length).toBe(2)
  expect(wrapper.find('.project-grid-tile').length).toBe(2)
})

it('<ProjectsOther> renders small view components correctly', () => {
  global.innerWidth = SMALL_WIDTH
  global.dispatchEvent(new Event('resize'))
  expect(wrapper.find('.projects-other').length).toBe(1)
  expect(wrapper.find('.title-text-other').length).toBe(1)
  expect(wrapper.find('.other-info').length).toBe(1)
  expect(wrapper.find('.description-other').length).toBe(1)
  expect(wrapper.find('.other-project-image').length).toBe(2)
  expect(wrapper.find('.project-title').length).toBe(2)
  expect(wrapper.find('.other-projects-small').length).toBe(2)
})

it('<ProjectsOther> opens link when gridtile is clicked in large view', () => {
  global.innerWidth = LARGE_WIDTH
  global.dispatchEvent(new Event('resize'))
  global.open = jest.fn()
  wrapper.find(`#other-project-${other[1].image}`).simulate('click')
  expect(global.open).toHaveBeenCalledWith(other[1].wiki)
})

it('<ProjectsOther> opens link when image is clicked in small view', () => {
  global.innerWidth = SMALL_WIDTH
  global.dispatchEvent(new Event('resize'))
  global.open = jest.fn()
  wrapper.find(`#other-project-small-${other[1].image}`).simulate('click')
  expect(global.open).toHaveBeenCalledWith(other[1].wiki)
})

// Snapshots
describe('<ProjectsOther> matches snapshot with:', () => {
  it('Large width', () => {
    global.innerWidth = LARGE_WIDTH
    global.dispatchEvent(new Event('resize'))
    const tree = renderer.create(
      <ProjectsOther projects={other} />,
    ).toJSON()
    expect(tree).toMatchSnapshot('ProjectsOtherLargelView.test.js.snap')
  })

  it('Small width', () => {
    global.innerWidth = SMALL_WIDTH
    global.dispatchEvent(new Event('resize'))
    const tree = renderer.create(
      <ProjectsOther projects={other} />,
    ).toJSON()
    expect(tree).toMatchSnapshot('ProjectsOtherSmallView.test.js.snap')
  })
})
