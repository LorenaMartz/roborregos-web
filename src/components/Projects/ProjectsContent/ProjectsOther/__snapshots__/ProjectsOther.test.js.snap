// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectsOther> matches snapshot with: Large width: ProjectsOtherLargelView.test.js.snap 1`] = `
<div
  className="projects-other"
>
  <div
    className="row"
  >
    <div
      className="other-info"
    >
      <h2
        className="title-text-other"
      >
        Other projects
      </h2>
      <div
        className="description-other"
      >
        <p>
          Check out other interesting tech projects developed by the community!
        </p>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="projects-other-grid"
    >
      <ul
        className="MuiGridList-root"
        style={
          Object {
            "margin": -1.5,
          }
        }
      >
        <li
          className="MuiGridListTile-root project-grid-tile"
          id="other-project-other_project1"
          onClick={[Function]}
          style={
            Object {
              "height": "auto",
              "padding": 1.5,
              "width": "33.333333333333336%",
            }
          }
        >
          <div
            className="MuiGridListTile-tile"
          >
            <div>
              <img
                alt="other_project1"
                className="other-project-image"
                src="other_project1.jpg"
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                  }
                }
              />
            </div>
            <div
              className="project-title"
            >
              <p>
                other_project1
              </p>
            </div>
          </div>
        </li>
        <li
          className="MuiGridListTile-root project-grid-tile"
          id="other-project-other_project2"
          onClick={[Function]}
          style={
            Object {
              "height": "auto",
              "padding": 1.5,
              "width": "33.333333333333336%",
            }
          }
        >
          <div
            className="MuiGridListTile-tile"
          >
            <div>
              <img
                alt="other_project2"
                className="other-project-image"
                src="other_project2.jpg"
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                  }
                }
              />
            </div>
            <div
              className="project-title"
            >
              <p>
                other_project2
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<ProjectsOther> matches snapshot with: Small width: ProjectsOtherSmallView.test.js.snap 1`] = `
<div
  className="projects-other"
>
  <div
    className="other-info"
  >
    <h2
      className="title-text-other"
    >
      Other projects
    </h2>
    <div
      className="description-other"
    >
      <p>
        Check out other interesting tech projects developed by the community!
      </p>
    </div>
  </div>
  <div>
    <div
      className="other-projects-small container"
      id="other-project-small-other_project1"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-7"
        >
          <img
            alt="other_project1"
            className="other-project-image"
            src="other_project1.jpg"
          />
        </div>
        <div
          className="col-5"
        >
          <div
            className="project-title"
            style={
              Object {
                "color": "black",
              }
            }
          >
            <p>
              other_project1
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="other-projects-small container"
      id="other-project-small-other_project2"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-5"
        >
          <div
            className="project-title"
            style={
              Object {
                "color": "black",
              }
            }
          >
            <p>
              other_project2
            </p>
          </div>
        </div>
        <div
          className="col-7"
        >
          <img
            alt="other_project2"
            className="other-project-image"
            src="other_project2.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Matches snapshot with: Large width: ProjectsOtherSmallView.test.js.snap 1`] = `
<div
  className="projects-other"
>
  <div
    className="row"
  >
    <div
      className="other-info"
    >
      <h2
        className="title-text-other"
      >
        Other projects
      </h2>
      <div
        className="description-other"
      >
        <p>
          Check out other interesting tech projects developed by the community!
        </p>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="projects-other-grid"
    >
      <ul
        className="MuiGridList-root"
        style={
          Object {
            "margin": -1.5,
          }
        }
      >
        <li
          className="MuiGridListTile-root project-grid-tile"
          id="other-project-other_project1"
          onClick={[Function]}
          style={
            Object {
              "height": "auto",
              "padding": 1.5,
              "width": "33.333333333333336%",
            }
          }
        >
          <div
            className="MuiGridListTile-tile"
          >
            <div>
              <img
                alt="other_project1"
                className="other-project-image"
                src="other_project1.jpg"
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                  }
                }
              />
            </div>
            <div
              className="project-title"
            >
              <p>
                other_project1
              </p>
            </div>
          </div>
        </li>
        <li
          className="MuiGridListTile-root project-grid-tile"
          id="other-project-other_project2"
          onClick={[Function]}
          style={
            Object {
              "height": "auto",
              "padding": 1.5,
              "width": "33.333333333333336%",
            }
          }
        >
          <div
            className="MuiGridListTile-tile"
          >
            <div>
              <img
                alt="other_project2"
                className="other-project-image"
                src="other_project2.jpg"
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                  }
                }
              />
            </div>
            <div
              className="project-title"
            >
              <p>
                other_project2
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Matches snapshot with: Small width: ProjectsOtherSmallView.test.js.snap 1`] = `
<div
  className="projects-other"
>
  <div
    className="other-info"
  >
    <h2
      className="title-text-other"
    >
      Other projects
    </h2>
    <div
      className="description-other"
    >
      <p>
        Check out other interesting tech projects developed by the community!
      </p>
    </div>
  </div>
  <div>
    <div
      className="other-projects-small container"
      id="other-project-small-other_project1"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-7"
        >
          <img
            alt="other_project1"
            className="other-project-image"
            src="other_project1.jpg"
          />
        </div>
        <div
          className="col-5"
        >
          <div
            className="project-title"
            style={
              Object {
                "color": "black",
              }
            }
          >
            <p>
              other_project1
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="other-projects-small container"
      id="other-project-small-other_project2"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-5"
        >
          <div
            className="project-title"
            style={
              Object {
                "color": "black",
              }
            }
          >
            <p>
              other_project2
            </p>
          </div>
        </div>
        <div
          className="col-7"
        >
          <img
            alt="other_project2"
            className="other-project-image"
            src="other_project2.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
